frontend/wallet/src/render/components/SendButton.re

```diff
@@ -0,0 +1,164 @@
+open Tc;
+
+module Styles = {
+  open Css;
+  let errorText =
+    merge([Theme.Text.Body.regular, style([color(Theme.Colors.roseBud)])]);
+};
+
+type unvalidatedModalState = {
+  fromStr: option(PublicKey.t),
+  toStr: string,
+  amountStr: string,
+  feeStr: string,
+  memoOpt: option(string),
+  errorOpt: option(string),
+};
+
+let emptyModal = activeWallet => {
+  fromStr: activeWallet,
+  toStr: "",
+  amountStr: "",
+  feeStr: "",
+  memoOpt: None,
+  errorOpt: None,
+};
+
+[@react.component]
+let make = (~wallets, ~onSubmit) => {
```

Comment:
okay hear me out -- I was staring at this for a bit, trying to trace the callbacks, when I thought of potentially a nice thing.

Try making 

```
onSubmit : state => ([> | `No_wallet | `No_destination | `Zero_fee], unit) Task.t
```

And then in onclick you can "attempt" the task. By reifying the errors here, you can handle the stringification of them outside of the logic too which is nice.

In general, if you ever have a callback that's called once at the end of the function you can use a Task instead.

Reply:
Also this could be a case where the state machine complexity is hitting the point where we want to useReducer, but it's unclear if this is maybe still simple enough we can live without it.